### PR TITLE
Test the interaction of getters and own properties on HTMLCollection.

### PR DESCRIPTION
**Pull request by Ms2ger**

### CODE REVIEW

**Comment by foolip** [comment]
@Ms2ger can you rebase this to trigger CI again? Results may have changed since January.

**Comment by Ms2ger** [comment]
@foolip done

**Comment by foolip** [comment]
Results at https://wpt.fyi/results/dom/collections/HTMLCollection-own-props.html?label=pr_head&max-count=1&pr=14972 look reasonable and reveal a lack of interoperability.

File: dom/collections/HTMLCollection-own-props.html [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+  if (name) {
+    element.id = name;
+  }
+  document.body.appendChild(element);
+  t.add_cleanup(function() { element.remove(); });
+  return element;
+}
+
+test(function() {
+  var name = "named", tag = "a";
```
Comment (foolip) [inline dom/collections/HTMLCollection-own-props.html]: I like how only single-character tag names are used in alphabetical order in this test :)

**Review by foolip — APPROVED** [review]

**Pull request merged**